hicommonwealth/commonwealth
# User-Onboarding: Show SSO email in welcome onboard modal + email notifications config

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6646

## "How We Fixed It"
N/A

## Test Plan

- Enable `FLAG_USER_ONBOARDING_ENABLED` in `.env`.
- Add `type='create-account'` to `<AuthModal/>` instance in `SubLayoutHeader.tsx` or apply this diff
```diff
diff --git a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
index 4564d26340..c58647b700 100644
--- a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
+++ b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
@@ -59,6 +59,7 @@ export const SublayoutHeader = ({
         open={isFeedbackModalOpen}
       />
       <AuthModal
+        type='create-account'
         onClose={() => setIsAuthModalOpen(false)}
         isOpen={isAuthModalOpen}
       />

```
- Create a new SSO account (google/discord/twitter/apple/github or email) and verify you see the welcome onboard modal show up. 
- Verify you see the email pre-filled with the email you used to signup on the SSO service and the email field is disabled.

## Deployment Plan
N/A

## Other Considerations
There is some bug in the backend that doesn't save the email from SSO in the DB (i am fixing that in https://github.com/hicommonwealth/commonwealth/issues/6752), so on new accounts, you won't be able to test it, but it will work if you have an SSO account older than 2-3 months (For this you'd have to statically open the welcome modal).
